### PR TITLE
Lowell add zero to month and Price Chopper Test

### LINKED ISSUE #109: lowell add lead zero to month

### PR DESCRIPTION
**Pull request by zpeyton**
closes #109

### CODE REVIEW

File: site-scrapers/LowellGeneral/index.js [hunk 1]
```diff
@@ -141,7 +141,8 @@ function parseAvailability(body) {
                     ].innerText.replace(/ {2,}/g, " ");
                     const thMonth = headerText.split(" ")[1];
                     const thDay = leadZero(headerText.split(" ")[2]);
-                    const [month, year] = getYearForMonth(thMonth);
+                    let [month, year] = getYearForMonth(thMonth);
```
Comment (livgust) [inline site-scrapers/LowellGeneral/index.js:144]: https://momentjs.com/docs/#/displaying/format/
Have you thought about using moment for this? MMM = the three-day short month string, so you wouldn't have to hard-code that, and you can also parse it and then spit it back out in the format you want (with padded 0s). LMK what you think.

**Review by livgust — CHANGES_REQUESTED** [review]

File: site-scrapers/LowellGeneral/index.js (hunk 1 above)
Comment (zpeyton) [reply site-scrapers/LowellGeneral/index.js:127]: Good call. Uploaded a fix for this using moment.

File: site-scrapers/LowellGeneral/index.js [hunk 2]
```diff
@@ -29,20 +26,11 @@ function buildTimeSlotsUrl(stationIdAttr) {
     const locationId = stationIdAttr.split("_")[7];
 
     // start yesterday since they are looking for midnight of a given day
+    const midnight = "+00%3A00%3A00";
     const now = new Date(new Date().getTime() - 86400 * 1000);
```
Comment (livgust) [inline site-scrapers/LowellGeneral/index.js:30]: 1) this isn't "now" - we should change the name
2) let's not mix Date() and moment() - we can use https://momentjscom.readthedocs.io/en/latest/moment/03-manipulating/02-subtract/ and do `moment().local().subtract(1, 'days')` to get yesterday.

File: site-scrapers/LowellGeneral/index.js [hunk 3]
```diff
@@ -29,20 +26,11 @@ function buildTimeSlotsUrl(stationIdAttr) {
...
-    const startDate =
-        [
-            now.getFullYear(),
-            leadZero(now.getMonth() + 1),
-            leadZero(now.getDate()),
-        ].join("-") + "+00%3A00%3A00";
+    const startDate = moment(now).format("YYYY-MM-DD") + midnight;
     const nowPlus45Days = new Date(now.getTime() + 45 * 86400 * 1000);
```
Comment (livgust) [inline site-scrapers/LowellGeneral/index.js:32]: same here - let's use moment and do moment().local().add(45, 'days') (also, `now` is actually yesterday so really this is wrongly 44 days from today)

**Review by livgust — CHANGES_REQUESTED** [review]
sorry, 2 nitpicks! then I'll approve, promise :D

File: site-scrapers/LowellGeneral/index.js (hunk 3 above)
Comment (zpeyton) [reply site-scrapers/LowellGeneral/index.js:32]: All set

File: site-scrapers/LowellGeneral/index.js (hunk 2 above)
Comment (zpeyton) [reply site-scrapers/LowellGeneral/index.js:30]: All set

File: test/PriceChopper.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const assert = require("assert");
```
Comment (tkjones117) [inline test/PriceChopper.js:1]: Not sure why this test is in this PR. it's fine now but break it out next time please, unless there's a good reason!

File: site-scrapers/LowellGeneral/functions.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        });
+    });
+}
+
+// Since the year is not available to us we need to calculate the year.
+// Using the current year is straight forward but the problem is
+// what happens when the month goes from Dec of the current year to Jan of
+// the next year. We add 1 to the current year if the passed month (ex. 'Jan')
+// is less than the current month. Ex. Jan 2022 < December 2021
+function getYearForMonth(passedMonthName) {
```
Comment (tkjones117) [inline site-scrapers/LowellGeneral/functions.js:179]: Nice, we can reuse this logic in other places in our codebase (not your responsibility). 
https://github.com/livgust/covid-vaccine-scrapers/blob/master/lib/RxTouch.js#L22

File: test/LowellGeneral.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                hasAvailability = parsedData.hasAvailability;
+            }
+            availability = groupAppointmentsByDate(
+                availability,
+                parsedData.availability
+            );
+        }
+
+        const result = { hasAvailability, availability };
+        //console.dir(result);
```
Comment (tkjones117) [inline test/LowellGeneral.js:49]: extra console.log

Comment (tkjones117) [reply test/LowellGeneral.js:49]: I'll approve once you remove this

Comment (zpeyton) [reply test/LowellGeneral.js]: Good catch. All set.

**Review by tkjones117 — APPROVED** [review]
lgtm!

**Review by harcod — APPROVED** [review]
OK. 
- scraper functioned as expected. (no appointments)
- 'npm run test' PASSED

**Comment by tkjones117** [comment]
> OK.
> 
> * scraper functioned as expected. (no appointments)
> * 'npm run test' PASSED

FYI @harcod in case you don't already know, you can see the output of the test runs here: https://github.com/livgust/covid-vaccine-scrapers/pull/108/checks?check_run_id=2034297650 (they're auto run on every new commit to the branch) - so you don't need to run them yourself! should save you 30 seconds :)

**Pull request merged**